sources/org.osbuild.inline

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/python3
+"""Source for binary data encoded inline in the manifest
+
+This source can be used to transport data in the source
+section of the manifest. Each resource is ascii-encoded
+in the `data` property, where the encoding is specified
+in the `encoding` property. The resources is content
+addressed via the hash value of the raw data before the
+ascii encoding. This hash value is verified after the
+resource is decoded and written to the store.
+"""
+
+
+import base64
+import contextlib
+import json
+import os
+import sys
+import subprocess
+import tempfile
+
+from typing import Dict
+
+
+SCHEMA = """
+"additionalProperties": false,
+"required": ["items"],
+"properties": {
+  "item": {
+    "description": "Inline data indexed by their checksum",
+    "type": "object",
+    "additionalProperties": false,
+    "patternProperties": {
+      "(md5|sha1|sha256|sha384|sha512):[0-9a-f]{5,64}": {
+        "type": "object",
+        "additionalProperties": false,
+        "required": ["method", "data"],
+        "properties": {
+          "encoding": {
+            "description": "The specific encoding of `data`",
+            "enum": ["base64"]
+          },
+          "data": {
+            "description": "The ascii encoded raw data",
+            "type": "string"
+          }
+        }
+      }
+    }
+  }
+}
+"""
+
+
+def verify_checksum(filename, checksum):
+    algorithm, checksum = checksum.split(":", 1)
+    if algorithm not in ("md5", "sha1", "sha256", "sha384", "sha512"):
```

Comment:
Shouldn't this be caught by the schema's `patternProperties`?
Might also make sense to check the length of the checksum based on the algorithm.  An invalid length will be caught by the validation below, but it might be worth avoiding doing the hashing to begin with.

Reply:
This is taking verbatim from the curl source, should probably move that into a utils package though.

Reply:
Fixed in both places